frontend-project/src/services/documentType.ts

```diff
@@ -0,0 +1,10 @@
+import smallEodSDK from '@/utils/sdk';
+
+export interface DocumentType {
+  name: string;
+}
+
+export const fetchDocumentType = async (id: number): Promise<DocumentType> => {
+  const response = await new smallEodSDK.DocumentTypesApi().documentTypesRead(id);
```

Comment:
cały czas nie ma @mateuszmagulski potrzeby, by zapisywać to w `response` i później od razu to `response` zwracać

Reply:
Gdy wbijam to od razu do return dostaję błąd: Redundant use of `await` on a return value

Reply:
Ja mam coś takiego:
```javascript
export const fetchDocumentType = async (id: number): Promise<DocumentType> => {
  return new smallEodSDK.DocumentTypesApi().documentTypesRead(id);
};
```
I nie widzę nigdzie błędu, który zgłaszasz.

Reply:
@mateuszmagulski funkcja asynchroniczna w każdym wypadku zwraca promise, więc awaitowanie promisa, żeby zwrócić go jako inny promise jest czynnością redundantną
a w skrócie, to nie dawaj awaita przy czymś co zwracasz (w node, w deno to działa trochę inaczej z tego co kojarzę ale to mówię tak raczej w ramach ciekawostki). nie wiem jak u Ciebie ale u mnie VSCode sam krzyczy o to (nie wiem czy to z pluginu czy domyślnie, a nie chce mi się w tej chwili sprawdzać :P)